Sources/MapboxDirections/DirectionsOptions.swift

```diff
@@ -122,10 +123,133 @@ open class DirectionsOptions: Codable {
+           let shapeFormat = RouteShapeFormat(rawValue: mappedValue) {
+            self.shapeFormat = shapeFormat
+        }
+        if let mappedValue = mappedQueryItems[CodingKeys.routeShapeResolution.stringValue],
+           let routeShapeResolution = RouteShapeResolution(rawValue: mappedValue) {
+            self.routeShapeResolution = routeShapeResolution
+        }
+        if mappedQueryItems[CodingKeys.includesSteps.stringValue] == "true" {
+            self.includesSteps = true
+        }
+        if let mappedValue = mappedQueryItems[CodingKeys.locale.stringValue] {
+            self.locale = Locale(identifier: mappedValue)
+        }
+        if mappedQueryItems[CodingKeys.includesSpokenInstructions.stringValue] == "true" {
+            self.includesSpokenInstructions = true
+        }
+        if let mappedValue = mappedQueryItems[CodingKeys.distanceMeasurementSystem.stringValue],
+           let measurementSystem = MeasurementSystem(rawValue: mappedValue) {
+            self.distanceMeasurementSystem = measurementSystem
+        }
+        if mappedQueryItems[CodingKeys.includesVisualInstructions.stringValue] == "true" {
+            self.includesVisualInstructions = true
+        }
+        if let mappedValue = mappedQueryItems[CodingKeys.attributeOptions.stringValue],
+           let attributeOptions = AttributeOptions(descriptions: mappedValue.components(separatedBy: ",")) {
+            self.attributeOptions = attributeOptions
+        }
+        if let mappedValue = mappedQueryItems["waypoints"] {
+            let indicies = mappedValue.components(separatedBy: ";").compactMap { Int($0) }
+            if !indicies.isEmpty {
+                waypoints.enumerated().forEach {
+                    $0.element.separatesLegs = indicies.contains($0.offset)
+                }
+            }
+        }
+        
+        let waypointsData = [mappedQueryItems["approaches"]?.components(separatedBy: ";"),
+                             mappedQueryItems["bearings"]?.components(separatedBy: ";"),
+                             mappedQueryItems["radiuses"]?.components(separatedBy: ";"),
+                             mappedQueryItems["waypoint_names"]?.components(separatedBy: ";"),
+                             mappedQueryItems["snapping_include_closures"]?.components(separatedBy: ";")
+        ] as [[String]?]
+        
+        let getElement: ((_ array: [String]?, _ index: Int) -> String?) = { array, index in
+            if array?.count ?? -1 > index {
+                return array?[index]
+            }
+            return nil
+        }
+        
+        waypoints.enumerated().forEach {
+            if let approach = getElement(waypointsData[0], $0.offset) {
+                $0.element.allowsArrivingOnOppositeSide = approach == "unrestricted" ? true : false
+            }
+            
+            if let descriptions = waypointsData[1]?[$0.offset].components(separatedBy: ",") {
```

Comment:
Have you thought about using `getElement` here and below?

Reply:
Yes, you are right. I forgot to spread this usage across all items. My bad.